gspread/worksheet.py

```diff
@@ -926,7 +812,21 @@ def get(
             self.spreadsheet_id, range_name, params=params
         )
 
-        return ValueRange.from_json(response)
+        vals_unfilled = ValueRange.from_json(response)
+        try:
+            vals = fill_gaps(vals_unfilled)
+        except KeyError:
+            vals = [[]]
+
+        if combine_merged_cells is True:
+            spreadsheet_meta = self.client.fetch_sheet_metadata(self.spreadsheet_id)
+            worksheet_meta = finditem(
+                lambda x: x["properties"]["title"] == self.title,
+                spreadsheet_meta["sheets"],
+            )
+            vals = combined_merge_values(worksheet_meta, vals)
+
+        return ValueRange.from_json(response, vals)
```

Comment:
it feels odd to get the values then rework the then extract them again using `ValueRange` but this time the response object is almose useless.

I beleive we can simplify it like this:

```python
vals = fill_gaps(response.get("values", [[]]))
```

then bellow force update the response object with the values reformated:

```python
response["values"] = vals
return ValueRange.from_json(response)
```

which removes the changes on the `ValueRange.from_json` method too.

what do you think ? :thinking: